forms-flow-web/src/components/FOI/customComponents/StateDropDown.js

```diff
@@ -136,50 +140,69 @@ const StateDropDown = ({
       case StateEnum.callforrecords.name.toLowerCase():
         if (_isMinistryCoordinator && personalRequest)
           return _stateList.callforrecordsforpersonal;
-        if(personalIAO && (requestDetails.bcgovcode.toLowerCase() === "mcf" || requestDetails.bcgovcode.toLowerCase() === "msd"))
-          return _stateList.callforrecordscfdmsdpersonal
+        if (
+          personalIAO &&
+          (requestDetails.bcgovcode.toLowerCase() === "mcf" ||
+            requestDetails.bcgovcode.toLowerCase() === "msd")
+        )
+          return _stateList.callforrecordscfdmsdpersonal;
         return _stateList.callforrecords;
       case StateEnum.tagging.name.toLowerCase():
-         return _stateList.tagging;
+        return _stateList.tagging;
       case StateEnum.readytoscan.name.toLowerCase():
-        return _stateList.readytoscan;     
+        return _stateList.readytoscan;
       case StateEnum.review.name.toLowerCase():
-        if(personalIAO && (requestDetails.bcgovcode.toLowerCase() === "mcf" || requestDetails.bcgovcode.toLowerCase() === "msd"))
-          return _stateList.reviewcfdmsdpersonal
+        if (
+          personalIAO &&
+          (requestDetails.bcgovcode.toLowerCase() === "mcf" ||
+            requestDetails.bcgovcode.toLowerCase() === "msd")
+        )
+          return _stateList.reviewcfdmsdpersonal;
         return _stateList.review;
       case StateEnum.onhold.name.toLowerCase():
+        if (previousState === StateEnum.response.name) {
+          return _stateList.onhold.filter(
+            (_state) =>
+              _state.status.toLowerCase() !==
+              StateEnum.callforrecords.name.toLowerCase()
+          );
+        } else if (previousState === StateEnum.feeassessed.name) {
+          return _stateList.onhold.filter(
+            (_state) =>
+              _state.status.toLowerCase() !==
+              StateEnum.response.name.toLowerCase()
+          );
+        }
```

Comment:
Updated the onhold statelist based on previousState condition. If previousState == Response then include response else include CFR.